db/migrations/20220413104717_goteo_new_faq.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+use Goteo\Model\Faq;
+
+/**
+ * Migration Task class.
+ */
+class GoteoNewFaq
+{
+  public function preUp()
+  {
+      // add the pre-migration code here
+  }
+
+  public function postUp()
+  {
+      // add the faq-migration code here
+      // Create default slugs for everybody
+      foreach(Faq::query("SELECT id,title FROM faq")->fetchAll(\PDO::FETCH_OBJ) as $faq) {
+        $slug = Faq::idealiza($faq->title, false, false, 150);
+        try {
+            // If duplicate, let it null
+            Faq::query("UPDATE faq SET slug=:slug WHERE id=:id", [':id' => $faq->id, ':slug' => $slug]);
+        } catch(\PDOException $e) {
+            //
+        }
```

Comment:
This UPDATE query looks important to me.

If by any chance, an exception is thrown, the query won't be executed and it might let the system in an inconsistent status (we might expect all FAQs to have a slug value).

Perhaps, we could write a log that indicates that there was an issue with this UPDATE query. And deal with it in order to fix it, and let every FAQ with a slug.

Reply:
@javicarrillo I think it was you who did this. I don't think the library we are using rolls back if the postup function does not work... But yes, at least we should log the errors.